core/build.gradle

```diff
@@ -194,3 +196,37 @@ if (tasks.findByName('trimShadedJar')) {
         dontnote
     }
 }
+
+class PublicSuffixesTask extends DefaultTask {
+    // Rebuild up to once in 7 days, unless cleaned.
+    @Input
+    final def days = (System.currentTimeMillis() / TimeUnit.DAYS.toMillis(7)).intValue()
+    @OutputFile
+    final def publicSuffixesFile = project.file("${project.projectDir}/src/main/resources/com/linecorp/armeria/public_suffixes.txt")
+
+    @TaskAction
+    def run() {
+        try {
+            def tempFile = File.createTempFile("public_suffixes", null)
+            def req = (HttpURLConnection) new URL('https://publicsuffix.org/list/public_suffix_list.dat').openConnection()
+            def res = req.getInputStream()
+            if (req.getResponseCode() != 200) {
+                println "received a non-200 response from publicsuffix.org: ${req.getResponseCode()}"
+                return
+            }
+            res.eachLine('UTF-8') {
+                if (!it.isEmpty() && !it.startsWith("//")) {
+                    tempFile.append(IDN.toASCII(it) + "\n")
+                }
+            }
```

Comment:
How about sorting it first, so that we are sure we will generate the same file even if `publicsuffix.org` doesn't provide a sorted list, just in case?